app/src/features/FhirResourceTree/useFhirResourceTreeData.ts

```diff
@@ -63,27 +66,58 @@ const useFhirResourceTreeData = (
   const [createAttribute] = useApiAttributesCreateMutation();
 
   const isLoading = isAttributesLoading && isStructureDefinitionLoading;
-  const nodeId = node?.id;
+  const nodeDefinition = node?.definitionNode.definition;
   const nodePath = node?.path;
 
+  // DefinitionNode tree building
   const data = useMemo(() => {
-    if (structureDefinition?.snapshot) {
-      const elementDefinitions = structureDefinition.snapshot.element;
+    // If we already have all the definition needed to populate the node children
+    if (node && node.children.length === 0 && options?.skip) {
+      return node.definitionNode;
+    }
+    // Else, we need to populate children DefinitionNode
+    else if (structureDefinition?.snapshot) {
+      let elementDefinitions = structureDefinition.snapshot.element;
+      // Prefix paths with node definition's path
+      if (nodeDefinition) {
+        elementDefinitions = elementDefinitions.map((elementDefinition) => ({
+          ...elementDefinition,
+          id: createElementDefinitionPathOrId(
+            nodeDefinition.id ?? "",
+            elementDefinition.id ?? ""
+          ),
+          path: createElementDefinitionPathOrId(
+            nodeDefinition.path ?? "",
+            elementDefinition.path ?? ""
+          ),
+        }));
+      }
       const elementDefinition = elementDefinitions[0];
```

Comment:
can you explain why we take the first element and/or find an appropriate name for this element?